x-pack/plugins/security_solution/public/common/hooks/use_invalid_filter_query.tsx

```diff
@@ -13,24 +13,35 @@ import { appActions } from '../store/app';
 import { useAppToasts } from './use_app_toasts';
 import { useDeepEqualSelector } from './use_selector';
 
+export const genHash = (message: string) =>
```

Comment:
`nit` : I know you just extraacted this code to make a reusable function, Could you please add the purpose of this hash. I am not sure I 100% understand what this bitwise operation is doing.

Thanks.

Reply:
the purpose of most hash string functions is to convert a string of variable size to a string of a fixed size for further comparison. Is this what you mean when asking about the purpose? Or do you mean like why do we need the hash here in the first place? If the latter then I don't know it's a good question to the author. Because from my observations you probably can go by without a hash and just compare the original message strings altogether.